### DISCUSSION TITLE
Welcome to lwpkt Discussions!

### DISCUSSION THREAD

**Discussion by MaJerle** [MaJerle/lwpkt#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by huseyink** [comment]
Hi Majerle,

The lwpkt_process function needs to be called periodically because of the wait data state to update the last rx time. I am using this library in an RTOS-based project and am triggering an event in the rx idle callback to call the lwpkt_process function. 

Therefore, my last rx time only updates when the packet is received as valid, which might cause a timeout error if the packet state is in progress.

I need to update the last rx time in the rx idle event manually. Briefly, what should I do if I don’t call the lwpkt_process function periodically in an RTOS-based project?

**Reply by MaJerle** [reply]
Whenever you call it, simply pass the current time into it. It is advised to call it more often than received data, to take care of structure state machine reset if there are missing bytes in communication, or if there is desync. This is why time is needed, in fact.

https://github.com/MaJerle/lwpkt/blob/develop/lwpkt/src/include/lwpkt/lwpkt_opt.h#L215

**Reply by huseyink** [reply]
I use this library in RTOS based project. So I don't call the lwpkt_process periodically. I call it when the RX thread flag is set then lwpkt_process handles the received bytes. Should I call this function periodically to update **last_rx_time** when lwpkt state is in **lwpktWAITDATA**?

**Reply by MaJerle** [reply]
If you don't call it periodically, then "timeout" feature may reset the packet state machine. This is only in the case you receive half of packet. If you always provide all the data, it may work.

It is better you call process periodically, as explained above.
